testing/mock/privval.go

```diff
@@ -35,6 +37,19 @@ func (pv PV) SignVote(chainID string, vote *tmproto.Vote) error {
 		return err
 	}
 	vote.Signature = sig
+
+	var extSig []byte
+	// We only sign vote extensions for non-nil precommits
+	if vote.Type == tmproto.PrecommitType && !tmtypes.ProtoBlockIDIsNil(&vote.BlockID) {
```

Comment:
copy/paste from cometbft, see #4015 

I believe the context of why we have a `mock/privval.go`:
- tendermint team wanted to drastically reduce API so they marked many many types as private
- we implemented a mock of the privval by copying their implementation
- the release line that made privval private got abandoned because it was broken
- now I believe the privval is public again